crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -329,7 +341,7 @@ pub unsafe trait QueryData: WorldQuery {
         fetch: &mut Self::Fetch<'w>,
         entity: Entity,
         table_row: TableRow,
-    ) -> Self::Item<'w, 's>;
+    ) -> Option<Self::Item<'w, 's>>;
```

Comment:
isn't this adding branching into every fetch? I suppose in the common cases it might get optimized out. We should check the assembly and see.